### ISSUE TITLE
Automatic screenshot tooling fails with some locales

### ISSUE THREAD

**Issue by loremattei** [wordpress-mobile/WordPress-iOS#9720]
The automatic screenshot tooling (based on a UI testing + fastlane setup) currently fails on some locales (fr-FR, pt-PT and sometimes a few others).
We need to fix them in order to move to a CI based tooling.

### Step to reproduce
1. Setup the automatic screenshot tool
2. Run using fastlane

### Expected behaviour
The tool should generate iPhone X, iPhone 8 Plus and Ipad 12.7'' screenshots in all the supported locales (list in the Snapfile).

### Actual behaviour
Screenshots for some locales/devices are not generated.

**Comment by jtreanor** [comment]
With my changes from https://github.com/wordpress-mobile/WordPress-iOS/pull/9795 included, running the entire snapshot passes for all locales except Russian. I am starting to dig into the reasons for that now. It seems to fail on log in.

It is certainly possible that other locales have flaky behaviour, though I have not yet encountered that.

**Comment by loremattei** [comment]
Awesome! It looks like the bad behaviour can be related to some weird localisation issue), so it may be possible that it doesn't appear on the same languages every time, since the localised strings change every release.

**Comment by jtreanor** [comment]
@loremattei Thats an interesting point. I'll see if I can shed more light on it.

**Comment by jtreanor** [comment]
It seems that the issue is more nuanced that I thought. Failures are not limited to Russian and are not consistent.

Here is an example of a failure I received just once on the `fr-FR` locale:
```
    t =    41.34s Type 'WordPress.com' into "usernameField" TextField
    t =    41.34s     Wait for org.wordpress to idle
    t =    41.60s     Find the "usernameField" TextField
    t =    41.82s     Synthesize event
12:51:22.254 xcodebuild[62511:2422880] <IDETestOperationCoordinator: 0x7fa05355b190>: parseConsoleOutputFromOriginalOutput: called, 146 bytes written out to StandardOutputAndStandardError
12:51:22.255 xcodebuild[62511:2422880] _XCT_testCaseDidFailForTestClass:WordPressScreenshotGeneration.WordPressScreenshotGeneration method:testGenerateScreenshots withMessage:failed: Timed out after waiting 1.0s for KeyEventCompleted after sending event for 'W'. file:<unknown> line:0
```

Another failure was a timeout on finding the close button in the post editor:
```
    t =   100.01s         Assertion Failure: WordPressScreenshotGeneration.swift:103: No matches found for Find: Elements matching predicate '"Azctec Editor Navigation Bar" IN identifiers' from input {(
    NavigationBar, 0x6080003993d0, traits: 35192962023424, identifier: 'Articles du blog'
```

These errors suggest to me that the problems are not related to any specific locale/device combinations but to general flakiness.

I will continue digging to find a pattern in the failures and get to the bottom of it.

**Comment by loremattei** [comment]
Fixed by #9912
